src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -271,10 +274,47 @@ private void EmitResource(ResourceSymbol resourceSymbol)
 
             var typeReference = EmitHelpers.GetTypeReference(resourceSymbol);
             var body = resourceSymbol.DeclaringResource.Value;
-            if (body is IfConditionSyntax ifCondition)
+
+
+            var ancestors = this.context.SemanticModel.ResourceAncestors.GetAncestors(resourceSymbol);
+
+            var conditions = new List<SyntaxBase>();
+            foreach (var ancestor in ancestors)
             {
-                body = ifCondition.Body;
-                this.emitter.EmitProperty("condition", ifCondition.ConditionExpression);
+                if (ancestor.DeclaringResource.Value is IfConditionSyntax ifCondition)
+                {
+                    conditions.Add(ifCondition.ConditionExpression);
+                }
+            }
+
+            // Unwrap the 'real' resource body if there's a condition
+            {
+                if (body is IfConditionSyntax ifCondition)
+                {
+                    body = ifCondition.Body;
+                    conditions.Add(ifCondition.ConditionExpression);
+                }
+            }
+
+            if (conditions.Count == 1)
+            {
+                this.emitter.EmitProperty("condition", conditions[0]);
+            }
+            else if (conditions.Count > 1)
+            {
+                var @operator = new BinaryOperationSyntax(
+                    conditions[0], 
+                    SyntaxFactory.CreateToken(TokenType.LogicalAnd, "&"),
```

Comment:
"&" -> "&&"

Reply:
Btw the second argument is no longer necessary. I've updated `SyntaxFactory.CreateToken` to resolve text based on `TokenType`.

Reply:
Awesome !